virtualization/windowscontainers/management/container_networking.md

```diff
@@ -235,13 +266,58 @@ C:\> docker run -it --network=MyTransparentNet --ip=10.80.123.32 windowsserverco
+
+### Docker Compose
+
+[Docker Compose](https://docs.docker.com/compose/overview/) can be used to define and configure container networks alongside the containers/services that will be using those networks. The Compose 'networks' key is used as the top-level key in defining the networks to which containers will be connected. For example, the syntax below defines the default NAT network created by Docker to be the 'default' network for all containers/services defined in a given Compose file.
+
+```none
+networks:
+ default:
+  external:
+   name: "nat"
+```
+For further information on defining/configuring container networks using Docker Compose, refer to the [Compose File reference](https://docs.docker.com/compose/compose-file/).
+
+### Service Discovery
+Built in to Docker is Service Discovery, which handles service registration and name to IP (DNS) mapping for containers and services; with service discovery, it is possible for all container endpoints to discover each other by name (either container name, or service name). This is particularly valuable in scaled-out scenarios, where multiple container endpoints are being used to define a single service. In such cases, service discovery makes it possible for a service to be considered a single entity regardless of how many containers it has running behind the scenes. For multi-container services, incoming network traffic is managed using a round-robin approach, by which DNS load balancing is used to uniformly distribute traffic across all container instances implementing a given service.
 
 ## Caveats and Gotchas
 
 ### Firewall
 
 The container host requires specific Firewall rules to be created to enable ICMP (Ping) and DHCP. ICMP and DHCP are required by Windows Server Containers to ping between two containers on the same host, and to receive dynamically assigned IP addresses through DHCP. In TP5, these rules will be created through the Install-ContainerHost.ps1 script. Post-TP5, these rules will be created automatically. All Firewall rules corresponding to NAT port forwarding rules will be created automatically and cleaned up when the container stops.
 
+### Existing vSwitch Blocking Transparent Network Creation
+
+When a transparent network is created, it is bound to the external network by a vSwitch which is in turn connected to an external network adapter on the container host. The vSwitch for the transparent network is either created or discovered (if pre-existing) by Docker. The vSwitch is then connected to the external network when the Windows Host Network Service (HNS) identifies the network adapter on the host with which to bind. With this in mind, *it is possible to run into problems with connecting the vSwitch to the host's external network adapter if you already have an external vSwitch created that is not visible to Docker.* 
```

Comment:
If an external vSwitch is already on the container host - created out-of-band using Hyper-V Manager or PowerShell - but not yet visible to Docker, restart the docker and HNS services.
Restart-Service Docker
Restart-Service HNS